src/jdk.jshell/share/classes/jdk/jshell/execution/LocalExecutionControl.java

```diff
@@ -98,13 +99,17 @@ private static byte[] instrument(byte[] classFile) {
                         ClassTransform.transformingMethodBodies(
                             CodeTransform.ofStateful(() -> {
                                 Set<Label> priorLabels = new HashSet<>();
+                                AtomicBoolean methodEntry = new AtomicBoolean();
```

Comment:
Instead of doing this hack, I think you should do so by overriding `CodeTransform.atStart` instead. Unfortunately we did not provide a utility for it like `endHandler`.

Reply:
Makes sense - thanks. We'll have to delambdify but the result is easier to understand. Updated in 1eae6ac3ed2.